### PR TITLE
Docs: Consistent example headings & text bat2 #5446

### PR DESCRIPTION
**Pull request by aubergine10**
Main focus of these commits:

1. Ensure consistent headings in Examples section

2. Ensure valid code fence captions to get stylised code blocks

Other issues in the existing docs were largely overlooked and will be dealt with in future updates.

@pedrottimark Can you check if this has worked as expected?

### CODE REVIEW

File: docs/rules/arrow-parens.md [hunk 1]
```diff
@@ -87,7 +87,7 @@ a.then((foo) => { if (true) {}; });
 
 #### If Statements
 
-One benefits of this option is that it prevents the incorrect use of arrow functions in conditionals:
+One of benefits of this option is that it prevents the incorrect use of arrow functions in conditionals:
```
Comment (aubergine10) [inline docs/rules/arrow-parens.md]: Already fixed this typo in a subsequent batch of updates, but currently struggling with current PR process so if possible please allow it to slip.

**Comment by aubergine10** [comment]
@ilyavolodin I hear you might be able to help me with my github woes? I've got several branches and want to send a PR for each, but for some reason github keeps merging any past PRs in to current PR (compare this PR to #6453). Any ideas?

**Comment by ilyavolodin** [comment]
@aubergine10 I can try:-) How do you update your local repository? Here's what you should do (assuming you are on master)
```sh
git remote add upstream git@github.com:eslint/eslint.git (or https if you don't have certificate setup)
git fetch --all --tag
git pull upstream/master
git push origin/master
git checkout -b your_new_branch_name
git add -A
git commit -m "Commit message for your checkin"
git push origin your_new_branch_name
create PR on github
```
If you want multiple commits per PR, then repeat last 3 steps (before creating PR on github). If you need to pull changes from eslint repository do the following on your_new_branch_name
```sh
git fetch --all --tag
git pull --rebase upstream master
git push origin your_new_branch_name
```

If you want one commit per PR, you have to create a new branch for each commit. Also it depends where you create branch from, you should always create your new branches from master, otherwise you will pull the changes from the branch you are currently on into a new branch.

For more specific help, try to catch me on gitter channel.

**Comment by aubergine10** [comment]
Awesome, thanks! I'll give that a try tomorrow - I suspect I may still run in to issues though, because I've already made a mess of my branches, etc. :/

**Comment by aubergine10** [comment]
Closing this PR - managed to dig out of the mess I was in using `git cherry-pick` and am currently creating new untangled branches to do PRs from. :)

**Pull request closed without merging**